cmake/modules/FindGnuLd.cmake

```diff
@@ -37,8 +42,11 @@ if(NOT EXISTS "${GNULD_LINKER}")
   endif()
 
   find_program(GNULD_LINKER ${CROSS_COMPILE}ld.bfd ${LD_SEARCH_PATH})
-  if(NOT GNULD_LINKER)
+  if(GNULD_LINKER)
+    set(GNULD_LINKER_IS_BFD TRUE)
+  else()
     find_program(GNULD_LINKER ${CROSS_COMPILE}ld ${LD_SEARCH_PATH})
+    set(GNULD_LINKER_IS_BFD FALSE)
```

Comment:
is that always true, or just true 99% of the time ?
If we start reporting this, then we should really be sure that it's correct.

In linux, there will almost always an `ld.bfd` and `ld` will usually be a link to the `ld.bfd` exe.
The `almost` is to take into account any obscure distro / home built toolchain where `ld` could be an `ld.bfd` without `ld.bfd` existing (or not in path).

But what about other OS'es and pre-built cross tool-chains ?
Are we sure all conforms to the `<triple>-ld.bfd` or could some be lazy and provide only `<triple>-ld`, for example on Windows where links (the unix way) are not supported.
Such that `<triple>-ld.exe` is in fact a bfd.

I agree that most likely your assumption is correct, but worried about edge-cases.

Reply:
This is more of a case where we can be certain that the linker is bfd compatible. We do ask the compilers for their preferred `ld.bfd` binary, and if they say there is one, we assume it is bfd compatible. This is simply a case where if we have any doubts, assume the linker is not bfd compatible.

Reply:
accepted